File: netgear/RAX5/usr/lib/lua/webGetFunc/wizard.lua

local M = {}

function M.getWizardRestoreConfigInProgress()
--[[
    if io.open("/data/wizardRestoreConfigInProgress") ~= nil then
        return "true"
    else
        return "false"
    end
  ]]
  local exec  = require "luci.util".exec
  local res = exec("puDataStr get wizardCheck RestoreConfigInProgress")
  if string.match(res,'true') ~= nil then
      return "true"
  else
      return "false"
  end
  
end

function M.getWizardFwUpgradeInProgress()
--[[
    if io.open("/data/wizardFwUpgradeInProgress") ~= nil then
        return "true"
    else
        return "false"
    end
  ]]
  local exec  = require "luci.util".exec
  local res = exec("puDataStr get wizardCheck FwUpgradeInProgress")
  if string.match(res,'true') ~= nil then
      return "true"
  else
      return "false"
  end
end

function M.getWizardWifiConfigured()
--[[
    if io.open("/data/wizardWifiConfigured") ~= nil then
        return "true"
    else
        return "false"
    end
  ]]
  local exec  = require "luci.util".exec
  local res = exec("puDataStr get wizardCheck WifiConfigured")
  if string.match(res,'true') ~= nil then
      return "true"
  else
      return "false"
  end
end
return M

File: netgear/RAXE300/webs/php/wizard.php
<?
include_once 'common_utils.php';

function getWizardRestoreConfigInProgress()
{
    if (file_exists("/data/wizardRestoreConfigInProgress")) {
        return "true"; 
    }
    else {
        return "false";
    }
}

function getWizardFwUpgradeInProgress()
{
    if (file_exists("/data/wizardFwUpgradeInProgress")) {
        return "true"; 
    }
    else {
        return "false";
    }
}

function getWizardWifiConfigured()
{
    if (file_exists("/data/wizardWifiConfigured")) {
        return "true"; 
    }
    else {
        return "false";
    }
}
?>
